pkgs/development/compilers/haxe/default.nix

```diff
@@ -24,6 +24,21 @@ stdenv.mkDerivation {
     install -vd "$out/bin" "$out/lib/haxe/std"
     install -vt "$out/bin" haxe haxelib
     cp -vr std "$out/lib/haxe"
+
+    # collapse HAXELIB_PATH
+    for name in $out/bin/haxe $out/bin/haxelib; do
+      wrapProgram $name \
+        --run 'if [ "$HAXELIB_PATH" != "''${HAXELIB_PATH/:/}" ]; then
+                 NEW_HAXELIB_PATH=$(${coreutils}/bin/mktemp -d)
+                 for libdir in $(IFS=:; echo $HAXELIB_PATH); do
+                   for lib in $libdir/*; do
+                     [ -e "$NEW_HAXELIB_PATH/$(${coreutils}/bin/basename "$lib")" ] || ${coreutils}/bin/ln -s --target-directory=$NEW_HAXELIB_PATH "$lib"
+                   done
+                 done
+                 export HAXELIB_PATH=$NEW_HAXELIB_PATH
+                 unset NEW_HAXELIB_PATH
+               fi'
```

Comment:
Could we not just build something like an explicit environment at build time like we have for [python](https://github.com/nixos-users/wiki/wiki/General-recommendations#package-simple-custom-scripts) or ruby (`bundlerEnv`)?